### PR TITLE
Guarantee ndarray contiguity

### PR DESCRIPTION
**Pull request by derwind**
### Summary

- This PR is to guarantee the contiguity of the memory layout of `numpy.ndarray` as an input of `AerStatevector`.

### Details and comments

- If non contiguous `numpy.ndarray` is input, an `AerStatevector` instance with unexpected `data` is created.

### CODE REVIEW

**Comment by derwind** [comment]
I have checked [one of the logs of unittest](https://github.com/Qiskit/qiskit-aer/actions/runs/3696680365/jobs/6260987461) and there are two tests failing related to `AerStatevector`. The first is `test_drawings`, which seems to be caused by `matplotlib` not being installed. Maybe `qiskit-terra[visualization]` is also needed in "Install deps" in `.github/workflows/tests.yml`. The second is `test_number_to_latex_terms`, which may need to be updated. qiskit-terra 0.21.2 passed the test, but 0.22.3 does not. I think the behavior has changed due to an update on the terra side.

File: qiskit_aer/quantum_info/states/aer_statevector.py [hunk 1]
```diff
@@ -134,6 +134,9 @@ def sample_memory(self, shots, qargs=None):
 
     @staticmethod
     def _from_ndarray(init_data, configs):
+        if not init_data.flags['C_CONTIGUOUS']:
+            raise AerError('initial data must have contiguous memory layout')
```
[the remark this answers is not in the document]
Comment (derwind) [reply qiskit_aer/quantum_info/states/aer_statevector.py:138]: I will discuss with @hhorii whether to add `copy=True` keyword and modify codes accordingly. I would like to add a release note after that.

[the remark this answers is not in the document]
Comment (hhorii) [reply qiskit_aer/quantum_info/states/aer_statevector.py:138]: I think `copy=True` keyword looks better than throwing exception. @jakelishman Thanks for your suggestion.

[the remark this answers is not in the document]
Comment (derwind) [reply qiskit_aer/quantum_info/states/aer_statevector.py:138]: I would like to add a NumPy-flavored `copy` option as an argument of `AerStatevector`.

**Comment by derwind** [comment]
Now we are in a position to pass all tests except Neko Integration.

**Review by hhorii — APPROVED** [review]

**Pull request merged**